packages/web-components/@sfx/sayt/test/sayt.test.ts

```diff
@@ -65,11 +72,244 @@ describe('Sayt Component', () => {
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should return true if event does not provide a searchbar ID', () => {
+      const searchbar = sayt.searchbar = 'some-searchbar-id';
+      const event = { detail: {} };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should return false if event provides the wrong searchbar ID', () => {
+      sayt.searchbar = 'correct-searchbar-id';
+      const event = { detail: { searchbar: 'wrong-searchbar-id' } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.false;
+    });
+
+    it('should return true if event specifies a searchbar but SAYT has none specified', () => {
+      sayt.searchbar = undefined;
+      const event = { detail: { searchbar: 'some-searchbar-id' } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should not error if passed an event without a detail attribute', () => {
+      const event = {};
+      const callback = () => sayt.isCorrectSayt(event);
+
+      expect(callback).to.not.throw();
+    });
+  });
+
   describe('render()', () => {
     it('should return an instance of TemplateResult', () => {
       const result = sayt.render();
 
       expect(result).to.be.an.instanceof(TemplateResult);
     });
   });
+
+  describe('processClick()', () => {
+    let node: any = 'some-node';
+    let event: Event;
+
+    beforeEach(() => {
+      event = {
+        target: node,
+      } as Event;
+      sayt.nodeInSearchBar = () => false;
+    });
+
+    it('should hide SAYT if the event target is nowhere relevant', () => {
+      sayt.contains = stub().returns(false);
+      sayt.hideSayt = spy();
+
+      sayt.processClick(event);
+
+      expect(sayt.contains).to.be.calledWith(node);
+      expect(sayt.hideSayt).to.be.called;
+    });
+
+    it('should not SAYT if the event target is contained by SAYT', () => {
+      sayt.contains = stub().returns(true);
+      sayt.hideSayt = spy();
+
+      sayt.processClick(event);
+
+      expect(sayt.contains).to.be.calledWith(node);
+      expect(sayt.hideSayt).to.not.be.called;
+    });
+
+    it('should not hide SAYT if the event target is the provided search box', () => {
+      sayt.contains = stub().returns(false);
+      sayt.nodeInSearchBar = stub().returns(true);
+      sayt.hideSayt = spy();
+
+      sayt.processClick(event);
+
+      expect(sayt.contains).to.be.calledWith(node);
+      expect(sayt.nodeInSearchBar).to.be.calledWith(node);
+      expect(sayt.hideSayt).to.not.be.called;
+    });
+  });
+
+  describe('nodeInSearchBar()', () => {
+    it('should return true if given node is contained in the search bar', () => {
+      const searchbar = {
+        contains: stub().returns(true),
+      };
+      const querySelector = stub(document, 'querySelector').callsFake(() => searchbar);
```

Comment:
This line will probably change with the corresponding suggestion in the code under test.

```suggestion
      const querySelector = stub(document, 'querySelector').returns(searchbar);
```